Kernel/VM/Space.cpp

```diff
@@ -158,6 +159,27 @@ Region* Space::find_region_containing(const Range& range)
     return nullptr;
 }
 
+Vector<Region*> Space::find_regions_intersecting(const Range& range)
```

Comment:
This would be better written as `template<typename Callback> void Space::for_each_region_intersecting(const Range&, Callback)`. That way we don't incur any heap allocations (which we do here due to the use of `Vector`.)

Reply:
In this particular case it can't be avoided: We first have to collect all relevant Regions, check whether they might cause `EINVAL` in munmap, and only afterwards deallocate them. One *could* replace that by calling `for_each_region_intersecting` twice, but that extra processing time seems silly to avoid a few bytes of temporary heap memory.

Reply:
I recieved this advice before, but we decided to loop over the data two times, to avoid returning an error half way through reallocation, which could leave the memory pages in a bad state

Reply:
Right, looping is performed twice anyway. Also `kmalloc()` memory, once owned, is cheap, but calling `kmalloc()` and `kfree()` is quite slow and both take a global spinlock for the duration.

Reply:
So do I need to change something?

If you are referring to this line
```cpp
// otherwise just split the regions and collect them for future mapping new_regions.append(space().split_region_around_range(*old_region, range_to_unmap));

```
split_region is stack allocated and guaranteed to be of size one, and the new_region vector is also stack allocated and has enough stack-capacity to hold both possible split regions.

Reply:
Implementing `for_each_region_intersecting` will be non-trivial if you want to be able to loop over it and remove from the underlying `NonnullOwnPtrVector<Region> m_regions` at the same time. @awesomekling

Reply:
So nothing to do for me here?